StoreApp/Util/WebMapper.cs

```diff
@@ -61,5 +64,109 @@ public ProductViewModel Map(ProductDTO productDTO)
+                string.IsNullOrEmpty(productXML["brand"].ToString()) || string.IsNullOrEmpty(productXML["producer"].ToString()))
+            {
+                return null;
+            }
+            product.Name = productXML["name"].InnerText;
+            product.Price = decimal.Parse(productXML["price"].InnerText);
+            product.Quantity = int.Parse(productXML["quantity"].InnerText);
+
+            var unit = unitService.GetAll().FirstOrDefault(u => u.Name.Equals(productXML["unit"].InnerText));
+            if (unit == null)
+            {
+                return null;
+            }
+            product.UnitId = unit.Id;
+
+            product.Picture = productXML["picture"]?.InnerText;
+            product.Quality = productXML["quality"]?.InnerText;
+            product.Enable = Convert.ToBoolean(productXML["enable"].InnerText);
+
+            var category = categoryService.GetAll().FirstOrDefault(c => c.Name.Equals(productXML["category"].InnerText));
+            if (category == null)
+            {
+                return null;
+            }
+            product.CategoryId = category.Id;
+
+            var brand = brandService.GetAll().FirstOrDefault(b => b.Name.Equals(productXML["brand"].InnerText));
+            if (brand == null)
+            {
+                return null;
+            }
+            product.BrandId = brand.Id;
+
+            var producer = producerService.GetAll().FirstOrDefault(p => p.Name.Equals(productXML["producer"].InnerText));
+            if (producer == null)
+            {
+                return null;
+            }
+            product.ProducerId = producer.Id;
+
+            return product;
+        }
+
+        public ProductDTO Map(ProductViewModel product)
+        {
+            if (string.IsNullOrEmpty(product.Name) ||
+                product.Quantity < 0 || product.Price < 0 ||
+                string.IsNullOrEmpty(product.Unit) || string.IsNullOrEmpty(product.Category) ||
+                string.IsNullOrEmpty(product.Brand) || string.IsNullOrEmpty(product.Producer))
+            {
+                return null;
+            }
+
+            UnitDTO unit = unitService.GetAll().FirstOrDefault(u => u.Name.Equals(product.Unit));
+
+            if (unit == null)
+            {
+                return null;
+            }
+
+            CategoryDTO category = categoryService.GetAll().FirstOrDefault(c => c.Name.Equals(product.Category));
+
+            if (category == null)
+            {
+                return null;
+            }
+
+            BrandDTO brand = brandService.GetAll().FirstOrDefault(b => b.Name.Equals(product.Brand));
+
+            if (brand == null)
+            {
+                return null;
+            }
+
+            ProducerDTO producer = producerService.GetAll().FirstOrDefault(p => p.Name.Equals(product.Producer));
+
+            if (producer == null)
+            {
+                return null;
+            }
```

Comment:
I suppose we can optimize this block of code.
Remove empty space by combining rows

Reply:
fixed